src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/TagHelperBlockRewriterTest/AllowsCompatibleTagStructures8.stree.txt

```diff
@@ -0,0 +1,15 @@
+RazorDocument - [0..15)::15 - [<input></input>]
+    MarkupBlock - [0..15)::15
+        MarkupElement - [0..15)::15
+            MarkupTagHelperElement - [0..7)::7 - input[StartTagOnly] - InputTagHelper1 - InputTagHelper2
+                MarkupTagHelperStartTag - [0..7)::7
+                    MarkupTextLiteral - [0..6)::6 - [<input] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                        OpenAngle;[<];
+                        Text;[input];
+                    MarkupTextLiteral - [6..7)::1 - [>] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                        CloseAngle;[>];
```

Comment:
You might notice the structure of `MarkupTagHelperStartTag` is not as good as `MarkupStartTag`. This is part of the tag helper cleanup that I was planning to do which I'll get to sometime soon. This shouldn't block the Components or VSCode work.